datasets/titanic/metadata.json

```diff
@@ -35,51 +75,52 @@
             "name": "name",
             "description": "Name of the passenger",
             "@type": "ml:Field",
-            "dataType": "csvw:string",
+            "dataType": "sc:Text",
             "source": "#{passengers-table/name}"
           },
           {
             "name": "gender",
             "description": "Gender of passenger (0: Male, 1: Female)",
             "@type": "ml:Field",
-            "dataType": "csvw:integer",
-            "source": "#{passengers-table/sex}"
+            "dataType": "sc:Integer",
+            "source": "#{passengers-table/sex}",
+            "references": "#{gender_enums}"
           },
           {
             "name": "age",
             "description": "Age of passenger at time of death.",
             "@type": "ml:Field",
-            "dataType": "csvw:integer",
+            "dataType": "sc:Integer",
             "source": "#{passengers-table/age}"
           },
           {
             "name": "survived",
             "description": "Survival status of passenger (0: Lost, 1: Saved)",
             "@type": "ml:Field",
-            "dataType": "csvw:integer",
+            "dataType": "sc:Integer",
             "source": "#{passengers-table/survived}"
           },
           {
             "name": "pclass",
             "description": "Passenger Class (1st/2nd/3rd)",
             "@type": "ml:Field",
-            "dataType": "csvw:integer",
+            "dataType": "sc:Integer",
             "source": "#{passengers-table/pclass}"
           },
           {
             "name": "cabin",
             "description": "Passenger cabin.",
             "@type": "ml:Field",
-            "dataType": "csvw:string",
+            "dataType": "sc:Text",
             "source": "#{passengers-table/cabin}"
           },
           {
             "name": "embarked",
             "description": "Port of Embarkation (C: Cherbourg, Q: Queenstown, S: Southampton, ?: Unknown).",
-            "@type": "City/name",
-            "dataType": "csvw:string",
-            "source": "#{passengers-table/embarked}"
-  
+            "@type": "ml:Field",
+            "dataType": "sc:Text",
+            "source": "#{passengers-table/embarked}",
+            "references": "#{ports_enums}"
```

Comment:
See earlier comment about `semanticType`. We could change `references` into `semanticType` here.
`"semanticType": "#{ports_enum}"` might be more descriptive. I think.

Reply:
Here `references` is the [existing mechanism](https://docs.google.com/document/d/1EDUbwwSeSUpRZWczy6FNuCXTbKh9e5H4xXMnqgoCRj8/edit#bookmark=id.hz571yc5evxj) to join two record sets, and here we are re-using this mechanism.

Using the joining mechanism "as it" has two implications I can think of:
  1. nothing in the current spec says the two `RecordSet`s have to define the same set of keys (there could be more keys defined in `ports_enums`, or less). This is not necessarily a good thing in that case, and to load the data, it would be good to know upfront the definitive number of enum values, or at least a close upper bound.
  2. The "join semantic" is not obvious: are we looking at two different genuine `RecordSet`s (like "movies" and "ratings" in [MovieLens example](https://docs.google.com/document/d/1EDUbwwSeSUpRZWczy6FNuCXTbKh9e5H4xXMnqgoCRj8/edit#heading=h.27t88xicctl), or are we defining the set of possible values?

So indeed, I agree that it might be opportune to create `possible_values` field, or to specify a "join semantic" with references.

@benjelloun what do you think?

Reply:
I considered allowing references to RecordSets as an allowed value for dataType... But I stopped short of doing that, because that would introduce RecordSets as a type definition mechanism. 

I think it's better to keep types and possible values separate, and "references" is essentially a foreign key (so, possible values) mechanism.

On your specific points:
Re: 1. Yes, that's correct, and I don't think that's a problem. The target of "references" defines the set of possible values, and the source may not use all of those. The fraction of values that are actually used is data-dependent, and could be captured by computed statistics, but I don't think it should be represented in the schema.
Re: 2. I think the reference in MovieLens has the same semantics as this one: movies in the Ratings table exist in the Movies table. The actual join / flattening is done elsewhere, by creating a new Movies+Ratings RecordSet that combines Movies and Ratings. The reference information is just useful to know how to do the joining.

So for me "references" really means "possible_values", aka, foreign key. I don't see why we would need two separate constructs here. Or am I missing something?

Reply:
Thanks for this explanation, I agree!